honeybadgermpc/mpc.py

```diff
@@ -92,47 +108,65 @@ def call_mixin(self, name, *args, **kwargs):
                 value_to_share = self.field.random()
 
             # 'S' is for single shares
-            self.send(j, ('S', shareid, value_to_share))
+            self.send(dest, ('S', shareid, value_to_share))
 
         # Set up the buffer of received shares
         share_buffer = [self._share_buffers[i][shareid] for i in range(self.N)]
 
         point = EvalPoint(self.field, self.N, use_fft=False)
 
         # Create polynomial that reconstructs the shared value by evaluating at 0
-        opening = robust_reconstruct(
+        p, _ = await robust_reconstruct(
             share_buffer, self.field, self.N, t, point)
-        self._openings[shareid] = opening
-
-        p, _ = await opening
 
         # Return reconstructed point
         return p(self.field(0))
 
-    def open_share_array(self, sharearray):
+    async def open_share_array(self, sharearray):
```

Comment:
Why is this change required?

Reply:
Shouldn't this return a `Future` which resolves to the opened values later on - as per our data flow?

Reply:
It keeps this in line with `open`. It was weird how one returned a future but the other didn't. 

Perhaps returning a `GFElementFuture` from `open` and a list of `GFElementFuture`s for `open_share_array` and making both of these non-async would be a better idea.